File: 백준/Silver/10799. 쇠막대기/쇠막대기.py
lst = input()
stack = []
ans = 0

for i in range(len(lst)):

    if lst[i] == "(":
        stack.append("(")
    else:
        if lst[i-1] == "(":
            stack.pop()
            ans += len(stack)
        else:
            stack.pop()
            ans += 1

print(ans)

File: 백준/Silver/10799. 쇠막대기/쇠막대기.cc
#include <bits/stdc++.h>
using namespace std;

int main () {
    ios::sync_with_stdio(0);
    cin.tie(0);
    stack<char> stk;
    int ans = 0;
    string s;
    cin >> s;


    for(int i = 0; i < s.size(); i++) {
        if(s[i] == '(') {
            stk.push(s[i]);
        } else if(s[i] == ')') {
            if(s[i-1] == '(') {
                stk.pop();
                ans += stk.size();
            } else if(s[i-1] == ')') {
                stk.pop();
                ans ++;
            }
        }
    }
    cout << ans;

}